ui/src/QrDataProvider.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { createContext, useContext, useReducer } from 'react';
+
+interface State {
+  qrCodes?: null | string[];
+  jws?: null | string;
+  setQrCodes: (arg: any) => any;
```

Comment:
Ca you type these to match the instances you have in `initialState`? e.g

    ()=>null